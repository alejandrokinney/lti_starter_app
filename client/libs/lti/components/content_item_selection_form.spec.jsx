import React from 'react';
import { render } from 'enzyme';
import ContentItemSelectionForm from './content_item_selection_form';

describe('Content Item Selection Form', () => {
  let result;
  let props;
  it('matches the snapshot', () => {
    props = {
      launchData: {
        testKey: 'testValue',
        testedKey: 'testValue1',

      },
      contentItemReturnURL: 'http://www.example.com',
    };
    result = render(<ContentItemSelectionForm {...props} />);
    expect(result).toMatchSnapshot();
  });
});
