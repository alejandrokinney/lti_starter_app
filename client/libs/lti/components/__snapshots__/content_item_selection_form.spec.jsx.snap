// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Item Selection Form matches the snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "action": "http://www.example.com",
      "enctype": "application/x-www-form-urlencoded",
      "method": "post",
      "style": "display:none;",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "name": "testKey",
          "type": "hidden",
          "value": "testValue",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "name": "testedKey",
            "type": "hidden",
            "value": "testValue1",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "type": "submit",
            },
            "children": Array [
              Object {
                "data": "Finish",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "type": undefined,
            },
            "x-attribsPrefix": Object {
              "type": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "name": undefined,
            "type": undefined,
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "name": undefined,
            "type": undefined,
            "value": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "name": undefined,
          "type": undefined,
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "name": undefined,
          "type": undefined,
          "value": undefined,
        },
      },
      Object {
        "attribs": Object {
          "name": "testedKey",
          "type": "hidden",
          "value": "testValue1",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "type": "submit",
          },
          "children": Array [
            Object {
              "data": "Finish",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "type": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "name": "testKey",
            "type": "hidden",
            "value": "testValue",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "name": undefined,
            "type": undefined,
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "name": undefined,
            "type": undefined,
            "value": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "name": undefined,
          "type": undefined,
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "name": undefined,
          "type": undefined,
          "value": undefined,
        },
      },
      Object {
        "attribs": Object {
          "type": "submit",
        },
        "children": Array [
          Object {
            "data": "Finish",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "name": "testedKey",
            "type": "hidden",
            "value": "testValue1",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "name": "testKey",
              "type": "hidden",
              "value": "testValue",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "name": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "name": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "name": undefined,
            "type": undefined,
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "name": undefined,
            "type": undefined,
            "value": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "type": undefined,
        },
        "x-attribsPrefix": Object {
          "type": undefined,
        },
      },
    ],
    "name": "form",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "action": undefined,
      "enctype": undefined,
      "method": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "action": undefined,
      "enctype": undefined,
      "method": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
